cli/src/semgrep/diff_scan_config.py

```diff
@@ -0,0 +1,70 @@
+"""
+Differential scan configuration for Pro inter-file analysis
+
+It's important to note that the targeting schemes differ among three scanning
+modes: pro (interfile) differential scanning, open-source (per-file)
+differential scanning, and whole scan. Here's a breakdown of their respective
+targeting requirements:
+
+Pro Diff Scan: In this mode, all input files are designated as "targets", and
+specifically, the files that have changed between the head and baseline commits
+are considered "diff targets".
+
+OSS Diff Scan: For OSS differential scanning, the files that have changed
+between the head and baseline commits serve as the primary "targets".
+
+Whole Scan: In the case of the whole scan, all input files are categorized as
+"targets" without any distinction based on commit changes.
```

Comment:
There are three kinds of scans here, but only two represented in the config below. It's not at all clear from reading the use sites that `is_diff_scan` implies Pro interfile. Could you clarify this, at least by renaming `DiffScan` etc. to `InterfileDiffScan`, or preferably by modeling the OSS diff scan with the same data structure?

Reply:
made an update to the PR. The `DiffScanConfig` has been renamed to `TargetModeConfig` and it now offers three options: "whole scan," "diff scan," and "pro diff scan."